nserror-kt/src/nativeInterop/cinterop/NSErrorKtRuntime.def

```diff
@@ -1,12 +1,54 @@
 ---
+
+#include "stdint.h"
+#include "stddef.h"
+
+// Based on Kotlin Native runtime
+
+typedef struct ObjHeader ObjHeader;
+
+typedef struct FrameOverlay {
+  struct FrameOverlay* previous;
+  int32_t parameters;
+  int32_t count;
+} FrameOverlay;
+
+typedef struct ObjHolder {
+    FrameOverlay frame_;
+    ObjHeader* obj_;
+} ObjHolder;
+
+extern void EnterFrame(ObjHeader** start, int parameters, int count) __attribute__((nothrow));
+extern void LeaveFrame(ObjHeader** start, int parameters, int count) __attribute__((nothrow));
+
+static void NSErrorKt_Kotlin_ObjCExport_ObjHolder_init(ObjHolder* this) {
+    this->obj_ = NULL;
+    EnterFrame((ObjHeader**) &(this->frame_), 0, sizeof(*this)/sizeof(void*));
+}
+
+static void NSErrorKt_Kotlin_ObjCExport_ObjHolder_deinit(ObjHolder* this) {
+    LeaveFrame((ObjHeader**) &(this->frame_), 0, sizeof(*this)/sizeof(void*));
+}
+
 extern void Kotlin_ObjCExport_RethrowExceptionAsNSError(void* exception, void* outError, const void** types);
 
+extern void* Kotlin_ObjCExport_NSErrorAsException(void* error, ObjHeader** OBJ_RESULT);
```

Comment:
Now that we have `ObjHeader` the return type (here and for the NSErrorKt one below) can be `ObjHeader*`, right? https://github.com/JetBrains/kotlin/blob/44ed2e94f57d63b380a81ad9dceba95b57f2f33d/kotlin-native/runtime/src/main/cpp/Memory.h#L169

Reply:
Yes, I missed that there was previously an "incorrect" return type. With C being C, it doesn't affect the functionality. But changing it would improve readability, so that's a good idea.